src/partial.lenses.js

```diff
@@ -1215,3 +1215,30 @@ export const json = /*#__PURE__*/(process.env.NODE_ENV === "production" ? I.id :
 export const seemsArrayLike = x =>
   x instanceof Object && (x = x.length, x === (x >> 0) && 0 <= x) ||
   I.isString(x)
+
+// Specialized Lens Creators
+
+const isArray = x =>
+  !(x instanceof Object) || Array.isArray(x)
+
+export const pointer = s => {
+  // See https://tools.ietf.org/html/rfc6901
+  const ts = s.split('/')
+  const n = ts.length
+  const isURI = '#' === ts[0]
+  for (let i=1; i<n; ++i) {
+    let t =
+      isURI
+        ? decodeURIComponent( ts[i] )
+        : ts[i]
```

Comment:
Hmm... `encodeURIComponent('/')` is `"%2F"` so shouldn't the decode need to be done before splitting?

I'm also pretty sure calling `decodeURIComponent` for each part separately would be slower than calling it once for the whole string before splitting.  That is especially because each call may allocate a new string.

Reply:
I'm pretty sure it needs to be done post-split. "/" is also a valid in an object key. One of the test keys has a "/" inside it.

Reply:
With a little more thought, I think you're right. I'll give it a shot. I had forgotten that the '/' has to be encoded as '~1' in the pointers.

Reply:
Actually, I think you are right.  From the examples I've seen it seems that they include `/` characters.  However, `decodeURIComponent('/')` is `'/'` so it is safe to just decode the whole string in one go before splitting.  Also, I think there is no need to explicitly drop the beginning `#` - the first element in the array after split can be just be ignored.

Reply:
The key is that the `'/'` has to be encoded in the pointers (both formats), anyway (as `'~1'`), so `decodeURIComponent` won't touch that, anyway. I made the change and committed. Also kept the leading `#`.